### ISSUE TITLE
gnt-node info doesn't complete node names

### ISSUE THREAD

**Issue by Ganeti-Issues-Migrator** [ganeti/ganeti#553]
Originally reported of Google Code with ID 497.

```
[gnt-eng-release.hot] root@wpeb9:~# gnt-node info node
[gnt-eng-release.hot] root@wpeb9:~# gnt-node info node.domain.com
Node name: ...
  primary ip: ...
  secondary ip: ...
  master candidate: True
  drained: False
  offline: False
  master_capable: True
  vm_capable: True
  primary for no instances
  secondary for instances:
    - ...
    - ...
  node parameters:
    - exclusive_storage: default (False)
    - oob_program: default ()
    - spindle_count: default (1)

```
Originally added on 2013-06-20 15:35:35 +0000 UTC.

**Comment by Ganeti-Issues-Migrator** [comment]
```
The proper fix is already included in 2.8

commit 91c1a26542a47307df47ca5710c6a0099d323110
Author: Iustin Pop <iustin@google.com>
Date:   Wed Feb 27 17:50:40 2013 +0100

    Fix node partial name matching in Haskell code
    
    This implements QffHostname and fixes the node listing (as well as
    export listing when filtering on node name).
    
    This bug was hidden by the fact that node listing with "gnt-node list
    aa" works if you don't have live queries (as it was originally), as
    the choosing of wanted nodes out of the config based on short names
    works. What didn't work was later post-filtering based on such short
    names (kind of duplicate, but that's how the code path is).
    
    By implementing QffHostname, we can have custom equality checks, like
    in the Python code. What I don't like is how convoluted the testing on
    various left/right combinations is, but I didn't find an easier way.
    
    The included unittest tests the partial filtering behaviour, and fails
    if the node name flag is not set to QffHostname.
    
    Signed-off-by: Iustin Pop <iustin@google.com>
    Reviewed-by: Guido Trotter <ultrotter@google.com>


And the workaround in 2.7 was un-done after its merge to 2.8 (c4e0d06 and 56bbc5).
```
Originally added on 2013-07-03 08:03:13 +0000 UTC.

Changed State: Fixed
Added to Milestone: Release2.8
Removed from Milestone: Release2.9

**Comment by Ganeti-Issues-Migrator** [comment]
```
-- Empty comment --
```
Originally added on 2013-10-30 09:30:16 +0000 UTC.

Changed State: Released
